src/ColorGradient.h

```diff
@@ -0,0 +1,35 @@
+#ifndef COLORGRADIENT_H
+#define COLORGRADIENT_H
+
+#include <vector>
+#include <stdexcept>
+#include "ColorGradientTypes.h"
+
+namespace frame_helper {
+
+class ColorGradient {
+
+    struct ColorPoint {     // Internal class used to store colors at different points in the gradient.
+        float r, g, b;      // Red, green and blue values of our color.
+        float val;          // Position of our color along the gradient (between 0 and 1).
+        ColorPoint(float red, float green, float blue, float value) :
+                r(red), g(green), b(blue), val(value) {
+        }
+    };
+
+private:
+    std::vector<ColorPoint> color;      // An array of color points in ascending value.
```

Comment:
I agree with you. Inserting into the middle of std::vector requires all elements after the insertion point are shuffled, while on std::list requires just to create a new node, store the value and connect the pointers. Done!